src/EditorFeatures/CSharpTest/Diagnostics/AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest.cs

```diff
@@ -15,5 +15,128 @@ public abstract partial class AbstractCSharpDiagnosticProviderBasedUserDiagnosti
+    {
+        public bool IsCompleted => default;
+        public TResult GetResult() => default;
+        public void OnCompleted(Action continuation) { }
+        public void UnsafeOnCompleted(Action continuation) { }
+    }
+}
+
+namespace System.Threading.Tasks
+{
+    using System.Runtime.CompilerServices;
+
+    [AsyncMethodBuilder(typeof(AsyncValueTaskMethodBuilder))]
+    public readonly struct ValueTask : IEquatable<ValueTask>
+    {
+        public ValueTask(Task task) {}
+        public ValueTask(IValueTaskSource source, short token) {}
+
+        public bool IsCompleted => default;
+        public bool IsCompletedSuccessfully => default;
+        public bool IsFaulted => default;
+        public bool IsCanceled => default;
+
+        public Task AsTask() => default;
+        public ConfiguredValueTaskAwaitable ConfigureAwait(bool continueOnCapturedContext) => default;
+        public override bool Equals(object obj) => default;
+        public bool Equals(ValueTask other) => default;
+        public ValueTaskAwaiter GetAwaiter() => default;
+        public override int GetHashCode() => default;
+        public ValueTask Preserve() => default;
+
+        public static bool operator ==(ValueTask left, ValueTask right) => default;
+        public static bool operator !=(ValueTask left, ValueTask right) => default;
+    }
+
+    [AsyncMethodBuilder(typeof(AsyncValueTaskMethodBuilder<>))]
+    public readonly struct ValueTask<TResult> : IEquatable<ValueTask<TResult>>
+    {
+        public ValueTask(TResult result) {}
+        public ValueTask(Task<TResult> task) {}
+        public ValueTask(IValueTaskSource<TResult> source, short token) {}
+
+        public bool IsFaulted => default;
+        public bool IsCompletedSuccessfully => default;
+        public bool IsCompleted => default;
+        public bool IsCanceled => default;
+        public TResult Result => default;
+
+        public Task<TResult> AsTask() => default;
+        public ConfiguredValueTaskAwaitable<TResult> ConfigureAwait(bool continueOnCapturedContext) => default;
+
+        public bool Equals(ValueTask<TResult> other) => default;
+        public override bool Equals(object obj) => default;
+        public ValueTaskAwaiter<TResult> GetAwaiter() => default;
+        public override int GetHashCode() => default;
+        public ValueTask<TResult> Preserve() => default;
+        public override string ToString() => default;
+        public static bool operator ==(ValueTask<TResult> left, ValueTask<TResult> right) => default;
+        public static bool operator !=(ValueTask<TResult> left, ValueTask<TResult> right) => default;
+    }
+}
+
+namespace System.Collections.Generic
+{
+    public interface IAsyncEnumerable<out T>
+    {
+        IAsyncEnumerator<T> GetAsyncEnumerator();
+    }
+
+    public interface IAsyncEnumerator<out T> : IAsyncDisposable
+    {
+        System.Threading.Tasks.ValueTask<bool> MoveNextAsync();
+        T Current { get; }
+    }
+}";
```

Comment:
this was hte minimal version i coudl get where everything worked.

Reply:
Is there a copy of this somewhere in the compiler layer for their own testing we can share?

Reply:
No, compiler has a copy that is very differnet and depends on hteir own facad libs unfortunately.